### PR TITLE
Js more conversions

### PR DESCRIPTION
**Pull request by ricardoquesada**

### CODE REVIEW

File: cocos/scripting/js-bindings/manual/ScriptingCore.cpp [hunk 1]
```diff
@@ -1589,32 +1589,40 @@ bool ScriptingCore::isObjectValid(JSContext *cx, uint32_t argc, jsval *vp)
...
-//    }
+    js_proxy_t* nproxy;
+    js_proxy_t* jsproxy;
+    void *ptr = (void*)ref;
+    nproxy = jsb_get_native_proxy(ptr);
+    if (nproxy) {
+        JSContext *cx = getGlobalContext();
+        // FIXME: Creating a RootedObject here is not needed.
+        // it is being created only because jsb_get_js_proxy() requires one
+        // but only the raw pointer is used in jsb_get_js_proxy()
```
Comment (pandamicro) [inline cocos/scripting/js-bindings/manual/ScriptingCore.cpp:1600]: This is for safety for passing js object pointers when calling a function, it's recommended to passing any js object pointers with rooted object.

File: cocos/scripting/js-bindings/manual/ScriptingCore.cpp [hunk 2]
```diff
@@ -1999,6 +2007,26 @@ JSObject* jsb_ref_create_jsobject(JSContext *cx, cocos2d::Ref *ref, js_type_clas
     return js_obj;
 }
 
+JSObject* jsb_ref_autoreleased_create_jsobject(JSContext *cx, cocos2d::Ref *ref, js_type_class_t *typeClass, const char* debug)
+{
+    //    JSB_AUTOCOMPARTMENT_WITH_GLOBAL_OBJCET
```
Comment (pandamicro) [inline cocos/scripting/js-bindings/manual/ScriptingCore.cpp:2012]: No need for auto compartment, we can remove this comment, there is no function call here

**Comment by pandamicro** [comment]
Hi, @ricardoquesada  the modifications are good. I want to know how are you gonna generate the auto bindings ? Check for `create` function name in sfunctions.c ?

File: cocos/scripting/js-bindings/manual/ScriptingCore.cpp (hunk 2 above)
Comment (ricardoquesada) [inline cocos/scripting/js-bindings/manual/ScriptingCore.cpp:2012]: ok. I'll remove it.

**Comment by ricardoquesada** [comment]
@pandamicro yes, I'm doing that. Here is the commit for that:
https://github.com/ricardoquesada/bindings-generator/commit/a8de796193c69dfa67cf06769060bb8947e483ef

**Comment by pandamicro** [comment]
Hi , @ricardoquesada I have commented in your commit https://github.com/ricardoquesada/bindings-generator/commit/a8de796193c69dfa67cf06769060bb8947e483ef#commitcomment-14806816, please take a look

**Comment by ricardoquesada** [comment]
@pandamicro thanks. I added your changes. Let me know if you think it is ready to be merged, or needs more changes. Thanks.

**Comment by pandamicro** [comment]
I think it's ok to be merged, though you need to update bindings generator ref to make jenkins pass

**Comment by ricardoquesada** [comment]
[ci rebuild]

**Comment by ricardoquesada** [comment]
Ok. I updated the bindings (I've just merged the bindings-generator PR)

**Comment by pandamicro** [comment]
[ci rebuild] only mac failed to finish

**Comment by ricardoquesada** [comment]
Build is green. Merging it. thanks.

**Pull request merged**